File: games/friendzone4/src/main/java/fr/purpletear/friendzone4/game/activities/load/NavigationHandler.java
package fr.purpletear.friendzone4.game.activities.load;

import android.content.Context;
import android.content.Intent;

import fr.purpletear.friendzone4.game.activities.audiocinematic.AudioCinematic;
import fr.purpletear.friendzone4.game.activities.end.End;
import fr.purpletear.friendzone4.game.activities.loading.Loading;
import fr.purpletear.friendzone4.game.activities.main.Main;
import fr.purpletear.friendzone4.game.activities.poetry.Poetry;
import fr.purpletear.friendzone4.game.activities.textcinematic.TextCinematic;


/**
 * The NavigationHandler handles the destination from the currentActivity.
 */
class NavigationHandler {

    /**
     * Determines the destination
     */
    private Navigation destination;

    public enum Navigation {
        MENU,
        GAME,
        AUDIOCINEMATIC,
        LOADING,
        TEXTCINEMATIC,
        VIDEOAD,
        INTERAD,
        POETRY,
        END
    }

    NavigationHandler() {
        destination = Navigation.GAME;
    }

    /**
     * Reloads the destionation
     *
     * @param n Navigation enum
     */
    public void to(Navigation n) {
        destination = n;
    }

    boolean toMenu() {
        return destination == Navigation.MENU;
    }

    /**
     * Returns the destination Intent
     *
     * @param c Context
     * @return Intent
     */
    Intent getIntent(Context c) {
        switch (destination) {
            case LOADING:
                return new Intent(c, Loading.class);
            case TEXTCINEMATIC:
                return new Intent(c, TextCinematic.class);
            case AUDIOCINEMATIC:
                return new Intent(c, AudioCinematic.class);
            case MENU:
                return null;
            case GAME:
                return new Intent(c, Main.class);
            case POETRY:
                return new Intent(c, Poetry.class);
            case END:
                return new Intent(c, End.class);
            default:
                throw new IllegalArgumentException();
        }
    }

    /**
     * @return the request code number using ordinals
     */
    int getRequestCode() {
        return destination.ordinal();
    }
}

File: games/friendzone1/src/main/java/fr/purpletear/friendzone/activities/load/NavigationHandler.kt
/*
 * Copyright (C) PurpleTear, Inc - All Rights Reserved
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Proprietary and confidential
 * Written by Hocine Belbouab <hbdeveloppeur@gmail.com>, 1/10/19 11:38 AM
 *
 */

package fr.purpletear.friendzone.activities.load

import android.content.Context
import android.content.Intent
import fr.purpletear.friendzone.activities.main.Main
import fr.purpletear.friendzone.activities.textcinematic.TextCinematic

/**
 * The NavigationHandler handles the destination from the currentActivity.
 */
internal class NavigationHandler {

    /**
     * Determines the destination
     */
    var destination: Navigation? = null
    private set

    /**
     * @return the request code number using ordinals
     */
    val requestCode: Int
        get() = destination!!.ordinal

    enum class Navigation {
        MENU,
        GAME,
        TEXTCINEMATIC,
        VIDEOAD,
        INTERSTITIAL
    }

    init {
        destination = Navigation.GAME
    }

    /**
     * Reloads the destionation
     *
     * @param n Navigation enum
     */
    fun to(n: Navigation) {
        destination = n
    }

    fun toMenu(): Boolean {
        return destination == Navigation.MENU
    }

    /**
     * Returns the destination Intent
     *
     * @param c Context
     * @return Intent
     */
    @Suppress("ConstantConditionIf")
    fun getIntent(c: Context): Intent? {
        val clss = when (destination) {
            Navigation.MENU -> return null
            Navigation.GAME -> Main::class.java
            Navigation.TEXTCINEMATIC -> TextCinematic::class.java
            else -> throw IllegalArgumentException("destination : $destination")
        }

        return Intent(c, clss)
    }
}
